pathspider/plugins/evilbit.py

```diff
@@ -0,0 +1,111 @@
+from scapy.all import IP  # pylint: disable=E061
+from scapy.all import IPv6  # pylint: disable=E061
+from scapy.all import UDP  # pylint: disable=E061
+from scapy.all import TCP  # pylint: disable=E061
+from scapy.all import DNS  # pylint: disable=E061
+from scapy.all import DNSQR  # pylint: disable=E061
+from scapy.all import RandShort  # pylint: disable=E061
+
+import pathspider
+from pathspider.base import PluggableSpider
+from pathspider.forge import ForgeSpider
+from pathspider.chains.dns import DNSChain
+from pathspider.chains.basic import BasicChain
+from pathspider.chains.tcp import TCPChain
+from pathspider.chains.evil import EvilChain
+from pathspider.chains.base import Chain
+from pathspider.chains.tcp import TCP_SYN, TCP_SA
+
+
+class EvilBit(ForgeSpider, PluggableSpider):
+
+    name = "evilbit"
+    description = "Evil bit connectivity testing (RFC 3514)"
+    #version = pathspider.__version__
+    chains = [BasicChain, DNSChain, TCPChain, EvilChain]
+    connect_supported = ["tcpsyn", "dnsudp"]
+    packets = 2
+
+    def forge(self, job, seq):
+        sport = 0
+        while sport < 1024:
+            sport = int(RandShort())
+        if self.args.connect == 'tcpsyn':
+            l4 = (TCP(sport=sport, dport=job['dp']))
+        if self.args.connect == 'dnsudp':
+            l4 = (UDP(sport=sport, dport=job['dp']) /
+                  DNS(qd=DNSQR(qname=job['domain'])))
+        if ':' in job['dip']:
+            ip = IPv6(src=self.source[1], dst=job['dip'])
+        else:
+            ip = IP(src=self.source[0], dst=job['dip'])
+        if seq == 1:
+            ip.flags = 'evil'
+        return ip/l4
+
+    def combine_flows(self, flows):
```

Comment:
Can you refactor this to have less branches? Make use of a private function (starting with a underscore) if needed.